libs/community/tests/unit_tests/document_loaders/test_recursive_url_loader.py

```diff
@@ -0,0 +1,99 @@
+from __future__ import annotations
+
+import inspect
+import uuid
+from types import TracebackType
+from typing import Any, Type
+
+import aiohttp
+import pytest
+import requests_mock
+
+from langchain_community.document_loaders.recursive_url_loader import RecursiveUrlLoader
+
+link_to_one_two = """
+<div><a href="/one">link_to_one</a></div>
+<div><a href="/two">link_to_two</a></div>
+"""
+link_to_three = '<div><a href="../three">link_to_three</a></div>'
+no_links = "<p>no links<p>"
+
+fake_url = f"https://{uuid.uuid4()}.com"
+URL_TO_HTML = {
+    fake_url: link_to_one_two,
+    f"{fake_url}/one": link_to_three,
+    f"{fake_url}/two": link_to_three,
+    f"{fake_url}/three": no_links,
+}
+
+
+class MockGet:
+    def __init__(self, url: str) -> None:
+        self._text = URL_TO_HTML[url]
+        self.headers: dict = {}
+
+    async def text(self) -> str:
+        return self._text
+
+    async def __aexit__(
+        self, exc_type: Type[BaseException], exc: BaseException, tb: TracebackType
+    ) -> None:
+        pass
+
+    async def __aenter__(self) -> MockGet:
+        return self
+
+
+@pytest.mark.parametrize(("max_depth", "expected_docs"), [(1, 1), (2, 3), (3, 4)])
+@pytest.mark.parametrize("use_async", [False, True])
+def test_lazy_load(
```

Comment:
@nfcampos any idea why this test is failing with use_async=True in CI but passing for me locally
```bash
max_depth = 2, expected_docs = 3, use_async = True

    @pytest.mark.parametrize(("max_depth", "expected_docs"), [(1, 1), (2, 3), (3, 4)])
    @pytest.mark.parametrize("use_async", [False, True])
    def test_lazy_load(
        mocker: Any, max_depth: int, expected_docs: int, use_async: bool
    ) -> None:
>       loader = RecursiveUrlLoader(fake_url, max_depth=max_depth, use_async=use_async)

tests/unit_tests/document_loaders/test_recursive_url_loader.py:52: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
langchain_community/document_loaders/recursive_url_loader.py:173: in __init__
    self._lock = asyncio.Lock() if self.use_async else None
/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/asyncio/locks.py:81: in __init__
    self._loop = events.get_event_loop()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <asyncio.unix_events._UnixDefaultEventLoopPolicy object at 0x7f7ecac39340>

    def get_event_loop(self):
        """Get the event loop for the current context.
    
        Returns an instance of EventLoop or raises an exception.
        """
        if (self._local._loop is None and
                not self._local._set_called and
                threading.current_thread() is threading.main_thread()):
            self.set_event_loop(self.new_event_loop())
    
        if self._local._loop is None:
>           raise RuntimeError('There is no current event loop in thread %r.'
                               % threading.current_thread().name)
E           RuntimeError: There is no current event loop in thread 'MainThread'.

/opt/hostedtoolcache/Python/3.9.19/x64/lib/python3.9/asyncio/events.py:642: RuntimeError
```

Reply:
and potentially only failing on second, third test cases with use_async=True but not the first